### ISSUE TITLE
Add the "{..}" construct to syntax lookup

### ISSUE THREAD

**Issue by not-rusty** [rescript-association/rescript-lang.org#652]
I have no idea what that actually does, I found it in some declaration files.

**Comment by ryyppy** [comment]
That is an `open object` type. They are mostly relevant for `external` usage where you'd receive a object of "any structure" and you'd then use it the way you think it should be used.

Example: 

```res
@val external someObj: {..} = "someObj"

someObj["name"] = "Patrick"
someObj["twitter"] = "@ryyppy"
```

[Playground Link](https://rescript-lang.org/try?version=v10.1.2&code=AINwhgNgBApgHgFxgJwHaSgZwPYFsYDyARgFYBcUA3gHTUC+UAvFAEQ77EksBQ37hpANot0+FgF0mrAApgEyAJYBjANY9+nYQgDuChEmQSpLYMgCeZgA6WzPIA)

**Comment by alarbada** [comment]
Mmm, so like an object map, kind of a { [ key: string]: any } in typescript?

Shouldn't then this be on the syntax lookup?

**Comment by ryyppy** [comment]
There's a few more subtle things to consider when using this type. (had to create a new issue to clarify some things [here](https://github.com/rescript-lang/rescript-compiler/issues/6267)).

It should be on the syntax lookup and most likely in the object / interop sections at some point.

**Comment by CarlOlson** [comment]
> Edit: I mean, these quirks are really difficult to search. I am partly not using rescript because of that (also I don't believe I'm the target audience).

If it helps. This is used very rarely, it has legitimate uses, but is generally an escape hatch.

The `{ [ key: string]: any }` type is closer to `Js.Dict.t<any>`, only there is no `any` type in ReScript. `Js.Json.t` is often used for any unknown data.

**Comment by ivan-demchenko** [comment]
Hi @ryyppy and @CarlOlson,
Actually, this escape hatch is used in a very important place (where it shouldn't be). I started a [thread here](https://forum.rescript-lang.org/t/implicit-type-conversion/4889) and an [issue here](https://github.com/rescript-lang/rescript-react/issues/105).
TL;DR: 
```rescript
<input
  onChange={evt => {
    let x = ReactEvent.Form.target(evt)["value"] // x is 'a
    setIntegerValue(_ => x) // x is int, should not happen!
  }}
/>
```
I agree with @alarbada, this looks like something that shouldn't even take place in a language like Rescript.

@ryyppy I think in my example the compiler should panic if there's no decoder in place because `evt.target.value` can be anything and will blow up at runtime. Rescript should protect from such runtime errors. It should require a decoder, akin to Elm.

**Comment by Zekt** [comment]
`{..}` is mentioned in the [React part of doc](https://github.com/rescript-lang/rescript-lang.org/blob/afc345f9e2d9cef0bae795ddd26af7bcc2b669bb/pages/docs/react/latest/styling.mdx?plain=1#L71) but [the linked Object page](https://github.com/rescript-lang/rescript-lang.org/blob/master/pages/docs/manual/v12.0.0/object.mdx) doesn't mention it, would be nice adding there.
